File: top_interview_questions/1_easy/1_array/rotate_array/solution.py
from typing import List


class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        length = len(nums)
        k = k % length

        nums_rotated = nums[length - k:] + nums[:length - k]

        for i in range(length):
            nums[i] = nums_rotated[i]

File: top_interview_questions/1_easy/1_array/rotate_array/solution.go
package main

import "fmt"

func main() {
	nums := []int{1, 2, 3, 4, 5, 6, 7}
	rotate(nums, 3)
	fmt.Println(nums) // expected [5,6,7,1,2,3,4]
}

func rotate(nums []int, k int) {
	length := len(nums)
	k = k % length
	rotated := nums[length-k:]

	rotated = append(rotated, nums[:length-k]...)

	for i := 0; i < length; i++ {
		nums[i] = rotated[i]
	}
}
